### PR TITLE
{bio}[foss/2017a] Trinity v2.6.5, Jellyfish v2.2.6, Salmon v0.9.1, ...

### PR DESCRIPTION
**Pull request by ebirn**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/SAMtools/SAMtools-1.6-foss-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Copyright:: Copyright 2012-2014 Uni.Lu/LCSB, NTUA
+# Authors::   Robert Schmidt <roschmidt@ohri.ca>, Cedric Laczny <cedric.laczny@uni.lu>, Fotis Georgatos <fotis@cern.ch>
+# License::   MIT/GPL
+# $Id$
+#
+# Modified by: Adam Huffman
+# The Francis Crick Institute
+#
+# Modified for version 1.4 by: Kurt Lust, UAntwerpen
+#
```
Comment (ebirn) [inline easybuild/easyconfigs/s/SAMtools/SAMtools-1.6-foss-2017a.eb]: oops, forgot to add to this. or should ew remove this header completely?

**Comment by timeu** [comment]
I created a new easyconfig file for version Trinity v2.8.3. Can I add this to this PR or should I create a new one ?

File: easybuild/easyconfigs/s/SAMtools/SAMtools-1.6-foss-2017a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SAMtools/SAMtools-1.6-foss-2017a.eb]: better leave the header in-place, to retain attribution

whether you add yourself or not, is up to you, we don't really have a policy on this...

**Comment by boegel** [comment]
@timeu Please open a new PR, I'll look into getting this one merged, sorry for overlooking it for so long...

File: easybuild/easyconfigs/h/HTSlib/HTSlib-1.6-foss-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+source_urls = ['https://github.com/samtools/%(namelower)s/releases/download/%(version)s/']
+sources = [SOURCELOWER_TAR_BZ2]
+checksums = ['9588be8be0c2390a87b7952d644e7a88bead2991b3468371347965f2e0504ccb']
+
+# cURL added for S3 support
+dependencies = [
+    ('zlib', '1.2.11'),
+    ('bzip2', '1.0.6', '-GCCcore-6.3.0', True),
+    ('XZ', '5.2.3', '-GCCcore-6.3.0', True),
+    ('cURL', '7.54.0', '-GCCcore-6.3.0', True),
```
Comment (boegel) [inline easybuild/easyconfigs/h/HTSlib/HTSlib-1.6-foss-2017a.eb]: @ebirn (cc @timeu) Please remove the hardcode subtoolchains:

```python
dependencies = [
    ('zlib', '1.2.11'),
    ('bzip2', '1.0.6'),
    ('XZ', '5.2.3'),
    ('cURL', '7.54.0'),
]

File: easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ transcript-level quantification estimates from RNA-seq data."""
+
+toolchain = {'name': 'foss', 'version': '2017a'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+source_urls = ['https://github.com/COMBINE-lab/%(namelower)s/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['3a32c28d217f8f0af411c77c04144b1fa4e6fd3c2f676661cc875123e4f53520']
+
+builddependencies = [('CMake', '3.8.1', '-GCCcore-6.3.0', True)]
```
Comment (boegel) [inline easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb]: @ebirn (cc @timeu) please remove hardcoding on subtoolchain

```python
builddependencies = [('CMake', '3.8.1')]

File: easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+dependencies = [
+    ('Boost', '1.65.1'),
+    ('tbb', '2017_U6', '-GCCcore-6.3.0', True),
```
Comment (boegel) [inline easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb]: @ebirn (cc @timeu) please remove hardcoding on subtoolchain

```python
('tbb', '2017_U6'),

File: easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+easyblock = 'ConfigureMake'
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: @ebirn (cc @timeu) Can you clarify why the existing `Trinity` easyblock is not used here?

Seems to work for Trinity 2.6.6, see #6160 which was merged recently

Comment (ebirn) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: I cannot. I can only say that this is probably due to the authors unawareness of that specific easyblock. 
Which really would have made this whole thing much easier.

I'm testing this now, modelling it more like the #6160

Comment (boegel) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: :+1:

File: easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: @timeu @ebirn Style nitpicking: please remove this empty line

File: easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+source_urls = ['https://github.com/COMBINE-lab/%(namelower)s/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['3a32c28d217f8f0af411c77c04144b1fa4e6fd3c2f676661cc875123e4f53520']
+
+builddependencies = [('CMake', '3.8.1')]
+
+dependencies = [
+    # this is to align dependencies with BLAST+
+    ('Boost', '1.63.0', '-Python-2.7.13'),
```
Comment (boegel) [inline easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb]: Please use `versionsuffix` here, like we do in existing `Salmon` easyconfigs (you'll need to rename the easyconfig file too):

```python
version = '0.9.1'
versionsuffix = '-Python-2.7.13'
...
dependencies = [
    # this is to align dependencies with BLAST+
    ('Boost', '1.63.0', versionsuffix),
...

File: easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    ('ant', '1.10.1', '-Java-%(javaver)s', True),
+    ('Bowtie', '1.2.2'),
+    ('Bowtie2', '2.3.2'),
+    ('ncurses', '6.0'),
+    ('zlib', '1.2.11'),
+    ('Perl', '5.24.1'),
+    ('SAMtools', '1.6'),
+    ('Jellyfish', '2.2.6'),
+    ('Salmon', '0.9.1'),
+    ('BLAST+', '2.6.0', '-Python-2.7.13'),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: @ebirn @timeu We don't have these 4 dependencies in our most recent `Trinity` easyconfig (`Trinity-2.6.6-intel-2018a.eb`).
Can you clarify why these have been added? Should we add them to `Trinity-2.6.6-intel-2018a.eb` as well (if so, we'll do that in a separate PR)?

Comment (ebirn) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: Yes I believe these tools should be added. They are at least optional dependencies:

1) SAMtools, see https://github.com/trinityrnaseq/trinityrnaseq/search?q=samtools&unscoped_q=samtools
more specifically, for example: direct call to samtools: https://github.com/trinityrnaseq/trinityrnaseq/blob/e678e74a99c3c1310968eb273d90ed259886f917/util/retrieve_sequences_from_fasta.pl#L14
2) Jellyfish: https://github.com/trinityrnaseq/trinityrnaseq/search?p=2&q=jellyfish&unscoped_q=jellyfish
called from Trinity directly, used in Dockerfile setup, as well as ugly Bioconda path replacement: https://github.com/trinityrnaseq/trinityrnaseq/blob/ec9896fcf422bc21b8558f13b0f8c2f80fa236f4/bioconda_recipe/trinity/2.6.6/build.sh
3) Salmon: https://github.com/trinityrnaseq/trinityrnaseq/search?q=salmon&unscoped_q=salmon
referenced in Changelog, code, Dockerfile, Travis env setup
4) BLAST+: various references: https://github.com/trinityrnaseq/trinityrnaseq/search?q=blast%2B&unscoped_q=blast%2B
also used in Dockerfile setup, even though I can't seem to find a reference in the other sources, where i.e. blastn is called directly

Maybe on 4) somebody else can chime in? I'm not 100% on this

File: easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb (hunk 6 above)
Comment (ebirn) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: will do.

File: easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb (hunk 7 above)
Comment (ebirn) [reply easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb]: I'll go through the dependencies and match them up. For consistency: when I add an versionsuffix, that would also have to show up in the .eb filename?

In that specific case, I was using the Boost+Python version, to have the same dependency as the BLAST+ dependency. 
Or should I better use BLAST+ without Python?

Comment (boegel) [reply easybuild/easyconfigs/s/Salmon/Salmon-0.9.1-foss-2017a.eb]: Only use Boost & BLAST+ with Python in versionsuffix if the Python bindings are required.

File: easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb (hunk 8 above)
Comment (boegel) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.6.5-foss-2017a.eb]: @ebirn Thanks for clarifying, it doesn't hurt to add these deps, indeed (even BLAST+, if you see hints about it being used).

Are you up for opening a PR to add these to `Trinity-2.6.6-intel-2018a.eb`?

**Comment by boegel** [comment]
Little hope in getting this merged due to lack of follow-up + old toolchain, so closing...
